linked.js

```diff
@@ -0,0 +1,161 @@
+class Node {
+    constructor(value = null, next = null) {
+        this.value = value;
+        this.next = next;
+    }
+}
+
+class linkedList {
+    constructor() {
+        // this.head = null;
+        // this.tail = null;
+        // this.next = null;
+    }
+    appendValue(value) {
+        if (!this.tail) {
+            this.tail = new Node(value);
+            this.head = this.tail;
+        }
+        else {
+            let oldTail = this.tail;
+            this.tail = new Node(value);
+            oldTail.next = this.tail;
+            // this.tail.prev = oldTail;
+        }
+    }
+    prependValue(value) {
+        if (!this.head) {
+            this.head = new Node(value);
+            this.tail = this.head;
+        }
+        else {
+            let oldhead = this.head;
+            this.head = new Node(value);
+            this.head.next = oldhead;
+            // oldhead.prev = this.head;
+        }
+    }
+    getSize() {
+        let currentNode = this.head;
+        let size = 0;
+        while (currentNode) {
+            currentNode = currentNode.next
+            size++
+        }
+        return size;
+    }
+    getHead() {
+        return this.head;
+    }
+    getTail() {
+        return this.tail
+    }
+    at(index) {
+        let currentNode = this.head;
+        let i = 0;
+        while (currentNode) {
+            if (i === index) {
+                return currentNode
```

Comment:
Consider returning the value the user gave not a the internal node construct. Unless this is meant to be a private function.

I see a node return value helps with L71, L81 ... but the better user expectation should come first.